src/constructs/static-site/static-site.ts

```diff
@@ -0,0 +1,322 @@
+  OriginAccessIdentity,
+  ViewerCertificate,
+  ViewerProtocolPolicy,
+} from "@aws-cdk/aws-cloudfront";
+import type { Behavior, CloudFrontWebDistributionProps } from "@aws-cdk/aws-cloudfront/lib/web-distribution";
+import { CanonicalUserPrincipal, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Bucket } from "@aws-cdk/aws-s3";
+import { Annotations, Arn, ArnFormat, CfnOutput, Duration, Token } from "@aws-cdk/core";
+import { Stage } from "../../constants";
+import type { GuDomainNameProps } from "../../types/domain-names";
+import { StageAwareValue } from "../../types/stage";
+import { GuAppAwareConstruct } from "../../utils/mixin/app-aware-construct";
+import { GuCertificate } from "../acm";
+import type { GuStack } from "../core";
+import { AppIdentity } from "../core/identity";
+import { GuCname } from "../dns";
+import { GuS3Bucket } from "../s3";
+
+const GLOBAL_AWS_REGION = "us-east-1";
+const CLOUDFRONT_CERTIFICATE_REQUIREMENT_MESSAGE = `CloudFront requires ACM certificates from the ${GLOBAL_AWS_REGION} region.`;
+const ACM_ARN_RESOURCE = "certificate";
+
+export interface GuStaticSiteProps
+  extends Omit<CloudFrontWebDistributionProps, "originConfigs" | "comment" | "viewerCertificate">,
+    AppIdentity {
+  /**
+   * Domain the CloudFront distribution should be aliased as.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-distributionconfig.html#cfn-cloudfront-distribution-distributionconfig-aliases
+   */
+  domainNameProps: GuDomainNameProps;
+
+  /**
+   * ARN of a pre-existing ACM certificate to attach to CloudFront. It must:
+   *   - exist in the us-east-1 region
+   *   - not have a hard coded account number
+   *
+   * If not provided, a new certificate is created. Note, this means the stack must be deployed in the us-east-1 region.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-viewercertificate.html#cfn-cloudfront-distribution-viewercertificate-acmcertificatearn
+   */
+  preExistingCertificateArn?: StageAwareValue<string>;
+
+  /**
+   * Name of a pre-existing S3 bucket holding the contents of the static site.
+   *
+   * If provided, a CloudFormation Output is created for the `CanonicalUser` to grant `s3:GetObject` permissions to.
+   * If not provided, a CloudFormation Output of the bucket name is created.
+   *
+   * @see https://github.com/awsdocs/amazon-s3-userguide/blob/d9fa2605ba253b3ca6ca1a83da5fa6b0b9b6debb/doc_source/s3-bucket-user-policy-specifying-principal-intro.md
+   *
+   */
+  preExistingOriginBucketName?: string;
+
+  /**
+   * Behaviours of the S3 bucket origin.
+   *
+   * Defaults to compress responses and redirect http to https.
+   */
+  behaviours?: Behavior[];
+
+  /**
+   * Flag to opt-out of automatically creating a [[ `GuCname` ]]. You'll have to create it yourself.
+   * This is useful if you want to control the TTL, which defaults to 1 hour.
```

Comment:
> This is useful if you want to control the TTL, which defaults to 1 hour.

If this is the case it might be better to allow users to pass in a custom TTL instead of opting out completely?

I guess there is another key use case for opting out too: if you are using Route53 to manage DNS?